The patch "wext: refactor" by Johannes Berg refactored
wext code so that new kernels no longer get the wext
handlers through struct netdevice, instead they get
it through the struct wiphy which is cfg80211 specific.

For old kernels this means you get not wext handlers
anymore when backporting code, this adds the wext handler
back to the netdevice wireless_handlers to let compat
users use wext again.

We only do this for kernels <= 2.6.32 as 2.6.33 will use
the struct wiphy wireless handler.

--- a/net/wireless/core.c
+++ b/net/wireless/core.c
@@ -701,6 +701,10 @@ static int cfg80211_netdev_notifier_call
 		wdev->sme_state = CFG80211_SME_IDLE;
 		mutex_unlock(&rdev->devlist_mtx);
 #ifdef CONFIG_CFG80211_WEXT
+#if (LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,32))
+		if (!dev->wireless_handlers)
+			dev->wireless_handlers = &cfg80211_wext_handler;
+#endif
 		wdev->wext.default_key = -1;
 		wdev->wext.default_mgmt_key = -1;
 		wdev->wext.connect.auth_type = NL80211_AUTHTYPE_AUTOMATIC;
